### PR TITLE
copy root fs into tmpfs

### PR DESCRIPTION
**Pull request by byxorna**
@roymarantz Looks like this works as expected. This should hopefully solve the issues we have been seeing where we fill up the overlay FS causing IO errors. This change moves the rootfs onto tmpfs when booting up.

cc @maddalab @Primer42 @defect 

screen shot 2015-02-05 at 10 14 02 pm

### CODE REVIEW

**Comment by byxorna** [comment]
Hmm... actually this may not be happy just yet. It appears the cleanup step isnt properly detaching the old loop devices. It looks like the script lazy unmounts them, but if whatever is holding them open never closes them, the mount will stick around forever. Or, its not actually booting with the toram option. Either way ;)

Still debugging

**Comment by byxorna** [comment]
screen shot 2015-02-06 at 10 59 34 am

Got it sorted out :)
Successfully running root out of tmpfs

File: bootcd/dmsquash-live-root.toram.sh [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+ 
+# If the kernel parameter toram is set, create a tmpfs device and copy the 
+# filesystem to it. Continue the boot process with this tmpfs device as
+# a writable root device.
+if [ -n "$toram" ] ; then
+    blocks=$( blockdev --getsz $BASE_LOOPDEV )
+ 
+    echo "Create tmpfs ($blocks blocks) for the root filesystem..."
+    mkdir -p /image
+    mount -n -t tmpfs -o nr_blocks=$blocks tmpfs /image
```
Comment (roymarantz) [inline bootcd/dmsquash-live-root.toram.sh]: do we want to resize root to something larger than $blocks?

File: bootcd/dmsquash-live-root.toram.sh [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+ 
+    ln -s $ROOTFS_LOOPDEV /dev/root
+    printf '/bin/mount -o rw %s %s\n' "$ROOTFS_LOOPDEV" "$NEWROOT" > /mount/01-$$-live.sh
+    exit 0
+fi
+ 
+if [ -b "$OSMIN_LOOPDEV" ]; then
+    # set up the devicemapper snapshot device, which will merge
+    # the normal live fs image, and the delta, into a minimzied fs image
+    if [ -z "$toram" ] ; then
```
Comment (roymarantz) [inline bootcd/dmsquash-live-root.toram.sh]: isn't this dead code since line 222 does an exist?

**Comment by roymarantz** [comment]
:100: my testing of the boot image with this works perfectly!

File: bootcd/dmsquash-live-root.toram.sh (hunk 1 above)
Comment (byxorna) [inline bootcd/dmsquash-live-root.toram.sh]: hmm... yea we could make it dynamic, perhaps a % of ram up to a max?
First pass though, this gets us out of the woods

File: bootcd/dmsquash-live-root.toram.sh (hunk 2 above)
Comment (byxorna) [inline bootcd/dmsquash-live-root.toram.sh]: 222 is in ```[[ -n $toram ]]```, so this is still active

**Comment by byxorna** [comment]
Sweet. @roymarantz mind if i merge?

**Comment by roymarantz** [comment]
Please hold on the merge till possible legal issue gets settled.

**Comment by roymarantz** [comment]
We are good to merge now.

From a comment on the blog post:
espebra Mod  Roy • an hour ago
Hi Roy,

There are no restrictions, so use it as needed. Hope it will provide value for you :)

**Comment by byxorna** [comment]
Ok, great! Thanks @roymarantz!

**Pull request merged**